File: onyx-database/src/com/onyx/exception/UnknownDatabaseException.java
package com.onyx.exception;

/**
 * Created by tosborn1 on 12/31/15.
 *
 */
public class UnknownDatabaseException extends EntityException
{
    @SuppressWarnings("unused")
    protected String cause = null;

    @SuppressWarnings("unused")
    public UnknownDatabaseException()
    {

    }

    @SuppressWarnings("unused")
    public UnknownDatabaseException(Exception e)
    {
        this.cause = e.getMessage();
    }
}

File: onyx-database/src/main/kotlin/com/onyx/exception/UnknownDatabaseException.kt
package com.onyx.exception

import java.io.PrintWriter
import java.io.StringWriter

/**
 * Created by Tim Osborn on 12/31/15.
 *
 */
class UnknownDatabaseException @JvmOverloads constructor(e: Exception? = null) : OnyxException() {
    private var unknownCause: String? = null
    private var stack:String = ""

    init {
        this.unknownCause = e?.message

        val stringWriter = StringWriter()
        val printWriter = PrintWriter(stringWriter)
        e?.printStackTrace(printWriter)
        this.stack = stringWriter.toString()
    }
}
